### ISSUE TITLE
iFrame/cross-origin Passkey/WebAuthN authentication broken (at least) on Firefox

### ISSUE THREAD

**Issue by lxgr** [bitwarden/clients#10930]
### Steps To Reproduce

This happens for any cross-origin WebAuthN authentication flow, as described e.g. here: https://medium.com/@corbado_tech/webauthn-iframes-integration-for-cross-origin-authentication-d483117cf437

Where I've personally reproduced is using a Link passkey used in the checkout flow at e.g. OpenAI, but I suspect it's a general problem in that Bitwarden only considers passkeys of the currently displayed top-level origin to be eligible for authentication, not those of iFrames embedded with the appropriate permission policy, as described in the linked blog post.

1. Register a passkey on link.com
2. Go to OpenAI and attempt to purchase a ChatGPT Plus subscription. Select "login with passkey" on Link if prompted.

### Expected Result

Bitwarden prompts usage of a passkey registered to the iframe's domain (assuming that domain's "Permissions-Policy" header also allows it).

### Actual Result

Bitwarden does not prompt at all, but the macOS native dialog. This then lets me use system-native passkeys (but I don't have any) or passkeys via Bluetooth/QR code, which I selected.

I was then actually able to use my passkey in Bitwarden iOS! So it seems like Firefox and macOS support that functionality correctly.

### Operating System

macOS

### Web Browser

Firefox

### Build Version

2024.8.1

### Issue Tracking Info

- [X] I understand that work is tracked outside of Github. A PR will be linked to this issue should one be opened to address it, but Bitwarden doesn't use fields like "assigned", "milestone", or "project" to track progress.

**Comment by JABirchall** [comment]
Do you get `Error: Could not establish connection. Receiving end does not exist. trigger-autofill-script-injection.js:5:20` logged in console?

**Comment by lxgr** [comment]
@JABirchall 
I sure do! Seems like this is a known thing then?

**Comment by JABirchall** [comment]
Yea, seems all autofills are broken.
#10814 #10815

**Comment by lxgr** [comment]
Oh, you're right, it doesn't even work for first-party WebAuthN anymore either 🤦‍♂️ Sorry, I should have checked that first!

Once that's fixed, I can check again to see if this particular use case also works (again).

**Comment by lxgr** [comment]
Ok, for whatever reason it's now semi-working, without me having changed anything on my system (same extension version, same Firefox version).

Now when I click "continue with Link", the macOS native WebAuthN UI still pops up once. But when I dismiss it, and then click "log in with a passkey", the Bitwarden UI does show up and lets me proceed!

What I suspect could be happening is that the iFrame first queries for discoverable credentials and finds none (falling through to the system UI), but then provides a credential ID (at the point I can click "log in with a passkey", my email address is displayed, so the iFrame probably also has the credential ID available by then), which Bitwarden then does recognize.

Maybe this is intended behavior for security reasons, but it's at least a discrepancy with the system implementation: When I add an iCloud passkey to Link, the sign-in does work at the first step (presumably the one using the discoverable credential).

tl;dr: I believe discoverable credentials do not work in iFrames at the moment, unlike for the Firefox native implementation (leveraging macOS's backend); non-discoverable ones do. I'll change the title to reflect this.

**Comment by cbbit** [comment]
Hi there,

Thank you for your report! It has been escalated for further investigation. If you have more information that can help us, please add it below. 

Thanks!

**Comment by closebot-bw** [comment]
⚠️ **Stale Issue Notice**

This issue has been automatically marked as stale due to inactivity. It will be closed in **2 weeks** (October 15, 2025) if no further activity occurs.

If this issue is still relevant and you would like to keep it open, please:
- Comment on this issue to show continued interest
- Provide any additional information or updates
- Confirm that the issue still exists in the latest version

Thank you for your contribution to this project! 🙏

**Comment by closebot-bw** [comment]
🔒 **Issue Closed Due to Inactivity**

This issue has been automatically closed due to lack of activity for an extended period. We periodically review and close inactive issues to help maintain our issue tracker and focus on current priorities.

**If this issue is still relevant:**
- Please create a new issue with updated information
- Include steps to reproduce the problem if it's a bug report
- Mention if this issue still occurs in the latest version

Thank you for your contribution to this project. Your feedback helps us improve! 🙏
